src/api/controllers/songController.ts

```diff
@@ -0,0 +1,128 @@
+import { Request, Response } from "express";
+import { NotFoundException } from "../models/exceptions/NotFoundException";
+import { SongFilter } from "../models/filters/songFilter";
+import { SongService } from "../services/songService";
+import { statusCodes } from "../utils/statusCodes";
+
+export class SongController {
+  songService: SongService;
+
+  constructor() {
+    this.songService = new SongService();
+  }
+
+  // If query params are provided, the following fuction will filter albums, otherwise it will return all
+  getSongs = async (request: Request, response: Response) => {
+    const band = request.query.band as string;
+    const album = request.query.album as string;
+
+    const songFilter = new SongFilter(band, album);
+
+    const songs = await this.songService.getSongs(songFilter);
+
+    if (songs.length > 0) {
+      return response.status(statusCodes.SUCCESS).json(songs);
+    }
+
+    return response.status(statusCodes.NOT_FOUND).json({
+      error: "No songs were found for the parameters provided",
+    });
+  };
+
+  createSong = async (request: Request, response: Response) => {
+    const validationResult = validateSongData(request);
+
+    if (validateSongData(request)) {
+      return response.status(statusCodes.BAD_REQUEST).json({
+        error: validationResult,
+      });
+    }
+
+    const songName = request.body.name as string;
+    const albumId = parseInt(request.body.albumId as string);
+
+    try {
+      const result = await this.songService.createSong(songName, albumId);
+      return response.status(statusCodes.CREATED).json(result);
+    } catch (error: any) {
+      if (error instanceof NotFoundException) {
+        return response
+          .status(statusCodes.NOT_FOUND)
+          .json({ error: error.message });
+      }
+      return response
+        .status(statusCodes.INTERNAL_SERVER_ERROR)
+        .json({ error: error.message });
+    }
+  };
+
+  updateSong = async (request: Request, response: Response) => {
+    const bodyMissing = checkRequestBody(request);
```

Comment:
Improve the checkRequestBody function name to be more descriptive. Also apply this same split approach to the AlbumController file